### PR TITLE
Add JS as a target for kmp-nativecoroutines-core

### PR DESCRIPTION
**Pull request by ankushg**
Part of https://github.com/rickclephas/KMP-NativeCoroutines/issues/17 and https://github.com/rickclephas/KMP-NativeCoroutines/pull/43

Doesn't add update any of KSP, but implements all `expect`s and gets tests passing for JS 😄

### CODE REVIEW

**Comment by ankushg** [comment]
Not sure if you want to merge this individually or wait until we add the logic to KSP

**Comment by ankushg** [comment]
@rickclephas Not sure if you want to merge this individually or wait until we add the logic to KSP

IMO merging it now makes sense so we can make sure things don't drift too much as other changes get merged

**Review by rickclephas — CHANGES_REQUESTED** [review]

File: kmp-nativecoroutines-core/src/jsTest/kotlin/com/rickclephas/kmp/nativecoroutines/RunTestJs.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+actual typealias TestResult = Promise<Unit>
+
+internal actual fun runTest(
+    context: CoroutineContext,
+    block: suspend CoroutineScope.() -> Unit
+): TestResult = kotlinx.coroutines.test.runTest(context = context, testBody = block)
+
+@Suppress("SuspendFunctionOnCoroutineScope")
+internal actual suspend fun CoroutineScope.runCurrent() {
+    coroutineContext[Job]?.children?.forEach { it.join() }
```
[the remark this answers is not in the document]
Comment (ankushg) [reply kmp-nativecoroutines-core/src/jsTest/kotlin/com/rickclephas/kmp/nativecoroutines/RunTestJs.kt]: Done (with the typecheck in `nativeCoroutinesMain`)

**Review by rickclephas — APPROVED** [review]

**Review by rickclephas — CHANGES_REQUESTED** [review]
Hmm looks like the coroutines-test classes aren't working as expected on JS 🤔

**Comment by ankushg** [comment]
> Hmm looks like the coroutines-test classes aren't working as expected on JS 🤔

Hmm it was green before switching JS from `coroutineContext[Job]?.children?.forEach { it.join() }` to `runCurrent`

**Comment by ankushg** [comment]
@rickclephas I figured it out! `runCurrent` runs the tasks that are pending **for the current virtual timestep**, but doesn't run any tasks that would be running in a future time step

What we really want to do is run **all** pending tasks until there are no more tasks to run. [`advanceUntilIdle` looks like it's designed to do that](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-test/kotlinx.coroutines.test/advance-until-idle.html) 😄 

Replacing `runCurrent()` with `advanceUntilIdle()` fixes the JS tests

**Review by rickclephas — APPROVED** [review]

**Pull request merged**